common/iterator/MappedIterator.java

```diff
@@ -44,4 +50,81 @@ public U next() {
     public void recycle() {
         iterator.recycle();
     }
+
+    /*
+    UNSAFE
+    The user must guarantee the mapping function preserves the sort order that the the source provides,
+    in the new domain.
+     */
+    static class Sorted<T extends Comparable<? super T>, U extends Comparable<? super U>> extends AbstractFunctionalIterator.Sorted<U> {
```

Comment:
Rather than saying `UNSAFE`, we should just describe the requirement and assert that `reverseMappingFn` is always symmetric to `mappingFn`, every time we compute `next`. (Not the other way around.)